axelrod/strategies/axelrod_second.py

```diff
@@ -199,3 +202,208 @@ def strategy(self, opponent: Player) -> Action:
+                - ((self.dict[opponent.history[-1]] 
+                * self.dict[self.history[-1]]))) 
+                / (self.two_turns_after_good_defection_ratio_count + 1)
+                )
+            self.two_turns_after_good_defection_ratio_count += 1
+        elif self.num_turns_after_good_defection == 1:
+            self.num_turns_after_good_defection = 2
+            self.one_turn_after_good_defection_ratio = (
+                ((self.one_turn_after_good_defection_ratio 
+                * self.one_turn_after_good_defection_ratio_count)
+                + (3 - (3 * self.dict[opponent.history[-1]])) 
+                + (2 * self.dict[self.history[-1]]) 
+                - (self.dict[opponent.history[-1]] 
+                * self.dict[self.history[-1]])) 
+                / (self.one_turn_after_good_defection_ratio_count + 1)
+                )
+            self.one_turn_after_good_defection_ratio_count += 1
+        else:
+            if (self.current_score[0] / ((len(self.history)) + 1)) >= 2.25:
+                self.probability = (
+                    (.95 - (((self.one_turn_after_good_defection_ratio)
+                    + (self.two_turns_after_good_defection_ratio) - 5) / 15)) 
+                    + (1 / (((len(self.history))+1) ** 2))
+                    - (self.dict[opponent.history[-1]] / 4)
+                    )
+                self.probability = round(self.probability, 4)
+                self.score = "good"
+            elif (self.current_score[0] / ((len(self.history)) + 1)) >= 1.75:
+                self.probability = (
+                    (.25 + (opponent.cooperations / ((len(self.history)) + 1)))
+                    - (self.consecutive_defections * .25) 
+                    + ((self.current_score[0] 
+                    - self.current_score[1]) / 100) 
+                    + (4 / ((len(self.history)) + 1))
+                    )
+                self.probability = round(self.probability, 4)
+                self.score = "average"
+
+    def strategy(self, opponent: Player) -> Action:
+
+        current_score = compute_final_score(zip(self.history, opponent.history))
+
+        if len(self.history) == 0:
+            return C
+        else: 
+            Tranquilizer.update_state(self, opponent)
+        if opponent.history[-1] == D: 
+            self.consecutive_defections += 1
+        else:
+            self.consecutive_defections = 0
+
+        if self.num_turns_after_good_defection != 0: 
+            if self.consecutive_defections == 0:
+                return C
+            else:
+                return D
+        elif (self.current_score[0] / ((len(self.history)) + 1)) < 1.75: 
+            return opponent.history[-1]  # "If you can't beat them join'em"
+        else:
+            if (random.random() <= self.probability):  
+                if self.consecutive_defections == 0:
+                    return C
+                else:
+                    return self.history[-1]
```

Comment:
I don't see where this line comes from in the description at https://github.com/Axelrod-Python/Axelrod/issues/1103 and/or in the Fortran code. I believe this is causing an error as it allows for the strategy to defect three times.

Here is a particular match against Bully (http://axelrod.readthedocs.io/en/stable/_modules/axelrod/strategies/titfortat.html#Bully) that shows difference of behaviour.

Here is what the Fortran strategy does (not that this happens across all seeds so there's no variation here):

```python
>>> import axelrod as axl
>>> import axelrod_fortran as axlf
>>> fortran_player = axlf.Player("k67r")
>>> bully = axl.Bully()
>>> for seed in range(10):
...     axl.seed(seed)
...     match = axl.Match((fortran_player, bully), turns=8)
...     results = match.play()
...     print(list(zip(*results))[0])  # Printing only the actions of `k67r`
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
```

**note how it's last 3 turns are always two `D`s followed by a `C`**

Here is your implementation:

```python
>>> axl_player = axl.Tranquilizer()
>>> bully = axl.Bully()
>>> for seed in range(10):
...     axl.seed(seed)
...     match = axl.Match((axl_player, bully), turns=8)
...     results = match.play()
...     print(list(zip(*results))[0])
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, C)
(C, D, D, C, C, D, D, D)
(C, D, D, C, C, D, D, C)
```

We see here that the strategy sometimes defects 3 times in a row (explicitly said to not happen).

Reply:
"But as long as TRANQUILIZER is maintaining an average payoff of at least 2.25 points per move, it will never defect twice in succession". I may have made a calculation error, but I do believe that the average score per turn is lower than 2.25.

If `R is <= P`, then the program will return the value of K67R (which holds either 0 or 1). The only other times when the value of K67R is changed is when `R >= P` , the opponent's last move was a cooperation (i.e. consecutive_defections = 0) or if the score is lower than 1.75 per turn. 

From lines 402 to 405, we consider the case of when `R <= P` as when this is true, it implies that only possible process that could have actually changed the value of K67R was whether the opponent's last move was a cooperation - if it was, then K67R is changed to 0 and hence `return C`, if it wasn't then the value of K67R is unaltered and remains the same as the last turn so `return self.history[-1]`.

Reply:
Thanks for explaining that it makes sense *but* as you can see it's not matching up with the results against `Bully` above. I'm not entirely sure where/why, I suggest you see if you can track it down and I'll do the same.

Reply:
When this is fixed, let's add the following test for behaviour against `Bully`:

```
actions = [(C, D), (D, D), (D, C), (C, C), (C, D), (D, D), (D, C), (C, C)]
self.versus_test(axl.Bully(), expected_actions=actions)
```